src/extensions/twowaycontactsyncadapter_impl.h

```diff
@@ -1372,7 +1381,15 @@ bool TwoWayContactSyncAdapter::purgeSyncStateData(const QString &accountId, bool
         purgeKeys << QStringLiteral("definitelyDownloadedAdditions");
     }
 
-    if (!d->m_engine->removeOOB(syncState.m_oobScope, purgeKeys)) {
+    // If this function is called without first calling initSyncAdapter, the
+    // OOB scope variable will not be initialized.  In that case, try to use
+    // the default OOB scope instead.
+    QString oobScope = syncState.m_oobScope;
+    if (oobScope.isEmpty()) {
```

Comment:
Hrm, well the unit test relies on being able to call this without having initialized the sync adapter.  But I guess I can modify the unit test to cache the information necessary to initialize, and then return an error here instead.  Will do.